sdk/keyvault/azure-security-keyvault-secrets/CMakeLists.txt

```
@@ -44,6 +44,24 @@ if(TSP_GEN)
     DownloadTSPFiles(${TSP_REPO} ${TSP_SHA} ${TSP_REPO_PATH}  ${TSP_DESTINATION})
 endif()
 
+option(AZURE_TSP_KV_SECRETS_GEN "Generate from TypeSpec" OFF)
+message("KeyVault Secrets TSP Generation ${az_TSP_KV_SECRETS_GEN}")
+if(AZURE_TSP_KV_SECRETS_GEN)
+    include(TSPCompile) 
+
+    set(TSP_REPO_PATH "specification/keyvault/data-plane/Security.KeyVault.Secrets/")
+    set(TSP_SHA "1124d1074d262dc20a010c2a9d366f8e7a452927")
+    set(TSP_DESTINATION "keyvault_secrets")
+
+    set(CODEGEN_SHA "b4107ff1b9ce83aedf1d299a63575b6c27434de0")
+    set(CODEGEN_DESTINATION "autorest.cpp")
```

We should drop references to autorest. This is typespec only.
```suggestion
    set(CODEGEN_DESTINATION "typespec-cpp")
```